### PR TITLE
Implement manual focus mode.

### PR DESCRIPTION
**Pull request by piggz**
The camera, while claiming to support manual appears to ignore it.  This
sets a flag to indicate manual mode, and sets the real mode to auto.  If
the flag is set, dont focus when pressing the shutter.  Reest the flag
when the mode is changed.

### CODE REVIEW

**Comment by llelectronics** [comment]
Works. 
Nice

File: qml/pages/CameraUI.qml [hunk 1]
```diff
@@ -358,19 +359,27 @@ Page {
...
+                camera.focus.setFocusMode(focus);
+                camera.start();
+            }
+        }
+        settings.mode.focus = focus;
 
-            //Set the focus point pack to centre
-            focusCircle.x = page.width / 2;
-            focusCircle.y = page.height / 2;
+        //Set the focus point pack to centre
```
Comment (mlehtima) [inline qml/pages/CameraUI.qml]: aWhile you are modifying this line fix the typo pack -> back.

File: qml/pages/CameraUI.qml [hunk 2]
```diff
@@ -358,19 +359,27 @@ Page {
     }
 
     function setFocusMode(focus) {
-        if (camera.focus.focusMode !== focus) {
+        if (focus === Camera.FocusManual && !_manualModeSelected) {
```
Comment (mlehtima) [inline qml/pages/CameraUI.qml]: The extra condition "!_manualModeSelected" in this causes focus mode to change to Auto if you press manual mode a second time in selection menu.

File: qml/pages/SettingsOverlay.qml [hunk 3]
```diff
@@ -183,7 +183,7 @@ Item {
     DockedListView {
         id: panelFocus
         model: modelFocus
-        selectedItem: settings.mode.focus
+        selectedItem: page._manualModeSelected ? Camera.FocusManual : camera.focus.focusMode
```
Comment (mlehtima) [inline qml/pages/SettingsOverlay.qml]: Is this change needed? You always have settings.mode.focus as the selected mode, in the new code you only set camera focus mode to Auto, not the settings mode.
